### PR TITLE
JAMES-2723 Move Mailbox and SimpleMailbox to mailbox-api

### PR DESCRIPTION
**Pull request by remk**

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/SimpleMailbox.java [hunk 1]
```diff
@@ -106,7 +104,7 @@ public MailboxPath generateAssociatedPath() {
     @Override
     public boolean equals(Object obj) {
         if (obj instanceof SimpleMailbox) {
-            SimpleMailbox o = (SimpleMailbox)obj;
+            SimpleMailbox o = (SimpleMailbox) obj;
```
[the remark this answers is not in the document]
Comment (remk) [reply mailbox/api/src/main/java/org/apache/james/mailbox/SimpleMailbox.java]: yes

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/Mailbox.java [hunk 2]
```diff
@@ -79,7 +76,7 @@
      */
     long getUidValidity();
     
-    
+
```
Comment (rouazana) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/Mailbox.java]: won't nitpick for this one ;)

**Comment by chibenwa** [comment]
>  Anybody knows enough about JPA to cut the link between JPAMailbox and Mailbox?
That way we would drop Mailbox interface entirely

Don't tempt me, I might give it a try!

**Comment by chibenwa** [comment]
Merged

**Comment by chibenwa** [comment]
>>  Anybody knows enough about JPA to cut the link between JPAMailbox and Mailbox?
That way we would drop Mailbox interface entirely

> Don't tempt me, I might give it a try!

https://github.com/linagora/james-project/pull/2292 did exactly this!

**Pull request closed without merging**